### PR TITLE
feat(dedicated): refactor network interfaces grid as component

### PR DESCRIPTION
**Pull request by darsene**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/multi-az_network-interfaces-refactor`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12285
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Refactoring of the datagrid network interfaces for dedicated servers as a component.
OLA and interfaces network APIs service split by feature (component)

### CODE REVIEW

**Review by qpavy — APPROVED** [review]

File: packages/manager/modules/bm-server-components/src/network-interfaces/attach/interfaces-attach.module.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
 import component from './interfaces-attach.component';
-import routing from './interfaces-attach.routing';
 
 const moduleName = 'ovhManagerDedicatedServerInterfacesAttach';
 
 angular
-  .module(moduleName, ['ui.router'])
-  .config(routing)
+  .module(moduleName, ['ovh-api-services'])
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/network-interfaces/attach/interfaces-attach.module.js:6]: ovh-api-services really needed ? Maybe you have the possibility to change the calls by $http and api endpoint ?

File: packages/manager/modules/bm-server-components/src/network-interfaces/detach/interfaces-detach.module.js [hunk 2]
```diff
@@ -1,11 +1,9 @@
 import component from './interfaces-detach.component';
-import routing from './interfaces-detach.routing';
 
 const moduleName = 'ovhManagerDedicatedServerInterfacesDetach';
 
 angular
-  .module(moduleName, ['ui.router'])
-  .config(routing)
+  .module(moduleName, ['ovh-api-services'])
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/network-interfaces/detach/interfaces-detach.module.js:6]: same https://github.com/ovh/manager/pull/10018#discussion_r1334512760

File: packages/manager/modules/bm-server-components/src/network-interfaces/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const moduleName = 'ovhManagerBmServerComponentsNetworkInterfacesComponent';
+
+angular
+  .module(moduleName, [
+    interfaceAttach,
+    interfaceDetach,
+    interfaceRename,
+    'oui',
+    'ui.router',
+    'ovh-api-services',
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/network-interfaces/index.js:22]: same https://github.com/ovh/manager/pull/10018#discussion_r1334512760

File: packages/manager/modules/bm-server-components/src/network-interfaces/rename/interfaces-rename.module.js [hunk 4]
```diff
@@ -1,11 +1,9 @@
 import component from './interfaces-rename.component';
-import routing from './interfaces-rename.routing';
 
 const moduleName = 'ovhManagerDedicatedServerInterfacesRename';
 
 angular
-  .module(moduleName, ['ui.router'])
-  .config(routing)
+  .module(moduleName, ['ovh-api-services'])
```
Comment (ghyenne) [inline packages/manager/modules/bm-server-components/src/network-interfaces/rename/interfaces-rename.module.js:6]: same https://github.com/ovh/manager/pull/10018#discussion_r1334512760

File: packages/manager/modules/bm-server-components/src/network-interfaces/attach/interfaces-attach.module.js (hunk 1 above)
Comment (darsene) [reply packages/manager/modules/bm-server-components/src/network-interfaces/attach/interfaces-attach.module.js:6]: yes, the controller uses  `OvhApiVrack`

File: packages/manager/modules/bm-server-components/src/network-interfaces/index.js (hunk 3 above)
Comment (darsene) [reply packages/manager/modules/bm-server-components/src/network-interfaces/index.js:22]: yes, the controller uses `OvhApiVrack`

File: packages/manager/modules/bm-server-components/src/network-interfaces/rename/interfaces-rename.module.js (hunk 4 above)
Comment (darsene) [reply packages/manager/modules/bm-server-components/src/network-interfaces/rename/interfaces-rename.module.js:6]: Yes the controller uses `OvhApiDedicatedServerVirtualInterface`

**Review by ghyenne — APPROVED** [review]

**Pull request merged**